messaging/src/androidTest/java/io/lantern/messaging/MessagingTest.kt

```diff
@@ -620,6 +621,120 @@ class MessagingTest : BaseMessagingTest() {
         }
     }
 
+    @Test
+    fun testMyself() {
+        testInCoroutine {
+            newDB.use { dogDB ->
+                newMessaging(dogDB, "dog").with { dog ->
+                    val dogId = dog.myId.id
+                    dog.addOrUpdateDirectContact(dogId, "Note to self")
+                    val msgs = sendAndVerify("dog sends note to dog", dog, dog, "hi myself")
+                    assertNotNull(msgs.received)
+                }
+            }
+        }
+    }
+
+    @Test
+    fun testMyselfAttachments() {
+        testInCoroutine {
+            newDB.use { dogDB ->
+                    newMessaging(dogDB, "dog").with { dog ->
+                            val dogId = dog.myId.id
+                            dog.addOrUpdateDirectContact(dogId, "Note to self")
+                            // lazy attachment to a non-existent file
+                            val badPlainTextFile = File(tempDir, UUID.randomUUID().toString())
+                            val badAttachment = dog.createAttachment(
+                                badPlainTextFile,
+                                "text/plain"
+                            )
+                            val lazyPlainTextFile = File(tempDir, UUID.randomUUID().toString())
+                            FileOutputStream(lazyPlainTextFile).use { output ->
+                                Util.copy(
+                                    ByteArrayInputStream(
+                                        "lazy attachment".toByteArray(
+                                            Charsets.UTF_8
+                                        )
+                                    ),
+                                    output
+                                )
+                            }
+                            val lazyAttachment = dog.createAttachment(
+                                lazyPlainTextFile,
+                                "text/plain"
+                            )
+                            val eagerPlainTextFile = File(tempDir, UUID.randomUUID().toString())
+                            FileOutputStream(eagerPlainTextFile).use { output ->
+                                Util.copy(
+                                    ByteArrayInputStream(
+                                        "eager attachment".toByteArray(
+                                            Charsets.UTF_8
+                                        )
+                                    ),
+                                    output
+                                )
+                            }
+                            val eagerAttachment = dog.createAttachment(
```

Comment:
I don't think you need to test all these different types of attachments. I think the only things you need to test are:

1. Eager attachments work
2. Lazy attachments work